### PR TITLE
Handle UTM/Attribution values in signup form

### PR DESCRIPTION
**Pull request by francoislehoux-okta**
## Description:
- **What's changed?** Handles UTM/Attribution values in signup form, reuses code from marketing websites

### Resolves:
* [OKTA-354076](https://oktainc.atlassian.net/browse/OKTA-354076)

Screen Shot 2021-01-08 at 2 54 50 PM
Screen Shot 2021-01-08 at 2 55 03 PM

### CODE REVIEW

File: packages/@okta/vuepress-theme-prose/index.js [hunk 1]
```diff
@@ -14,9 +14,10 @@ module.exports = (options, ctx) => {
       $page.fullHeaders = resolveHeaders($page);
 
       // Set to true to enable the redesign in development
-      const redesign = process.env.REDESIGN || false;
+      // const redesign = process.env.REDESIGN || false;
       // Respect the redesign flag unless this is a production build
-      $page.redesign = !ctx.isProd && redesign;
+      // $page.redesign = !ctx.isProd && redesign;
+      $page.redesign = true;
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/index.js:20]: note to revert before merging

File: packages/@okta/vuepress-theme-prose/util/attribution/cookies.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import Cookies from "cookies.js";
+
+const COOKIE_PREFIX = "_okta_";
+const HOST = "okta.com";
+const HOST_LENGTH = 8;
+
+function getDomain(host) {
+  return host.indexOf(HOST) == host.length - HOST_LENGTH && host;
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/util/attribution/cookies.js:8]: nit for readability:
```suggestion
  return (host.indexOf(HOST) == host.length - HOST_LENGTH) && host;
```

File: packages/@okta/vuepress-theme-prose/util/attribution/cookies.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+export function deleteCookie(key) {
+  const { host } = location;
+  const domain = getDomain(host);
+
+  return Cookies.remove(`${COOKIE_PREFIX}${key}`, { domain: domain || host });
+}
+
+export function getCookie(key, defaultValue = false) {
+  return (
+    JSON.parse(Cookies.get(`${COOKIE_PREFIX}${key}`) || "false") || defaultValue
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/util/attribution/cookies.js:20]: Suggest we break this into multiple statements for clarity

File: packages/@okta/vuepress-theme-prose/util/attribution/attribution.js [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+function setHiddenUtmValues(form) {
+  onAttach();
+
+  const attribution = getAttribution();
+
+  if (attribution) {
+    let paramValues;
+
+    for (let i = 0; i < acceptedParams.length; i++) {
+      let param = acceptedParams[i];
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/util/attribution/attribution.js:214]: Do we have access to `Array.prototype.forEach`?

File: packages/@okta/vuepress-theme-prose/util/attribution/attribution.js [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+      [].slice.call(document.querySelectorAll(`[name=${key}__c]`))
+    );
+
+    fieldExistsInForm =
+      $form &&
+      (!!$form.querySelectorAll(`[name=${key}]`).length ||
+        !!$form.querySelectorAll(`[name=${key}__c]`).length);
+
+    for (let i = 0; i < $fields.length; i++) {
+      let $el = $fields[i];
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/util/attribution/attribution.js:134]: Same question about if we have `forEach` here and in other locations where we have a `for` loop in this file.

File: packages/@okta/vuepress-theme-prose/package.json [hunk 6]
```diff
@@ -25,8 +25,11 @@
   "dependencies": {
     "@okta/odyssey": "^0.0.1",
     "axios": "^0.21.0",
+    "cookies.js": "^2.1.15",
     "include-media": "^1.4.9",
     "moment": "^2.24.0",
+    "moment-timezone": "^0.5.32",
+    "query-string": "^6.13.8",
```
Comment (francoislehoux-okta) [inline packages/@okta/vuepress-theme-prose/package.json:32]: This adds a bunch of libs but wanted to keep code as is for now, should def revisit for phase 2, including maybe replacing `moment` and `moment-timezone` with something lighter like `date-fns`

File: packages/@okta/vuepress-theme-prose/index.js (hunk 1 above)
Comment (francoislehoux-okta) [inline packages/@okta/vuepress-theme-prose/index.js:20]: **Cleanup**

File: packages/@okta/vuepress-theme-prose/util/attribution/attribution.js [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+}
+
+/**
+ * onAttach function.
+ *
+ * @access public
+ * @return void
+ */
+function onAttach() {
+  if (isAttached == false) {
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/util/attribution/attribution.js:64]: nit: strong preference against `==`
```suggestion
  if (isAttached === false) {
```

File: packages/@okta/vuepress-theme-prose/util/attribution/attribution.js [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+    .reduce((object, key) => {
+      object[key] = params[key];
+      return object;
+    }, {});
+
+  params = Object.assign(
+    {
+      utm_page: location.pathname,
+      utm_date: moment(moment.utc())
+        .tz("America/Los_Angeles")
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/util/attribution/attribution.js:48]: Is this just the TZ we already use in the existing code?

**Review by shawyu-okta — COMMENTED** [review]
Feel free to resolve/ignore comments that are on code that was just copied from the other site where appropriate

File: packages/@okta/vuepress-theme-prose/util/attribution/attribution.js (hunk 8 above)
Comment (francoislehoux-okta) [reply packages/@okta/vuepress-theme-prose/util/attribution/attribution.js:48]: We use moment to compute the "last updated" and it most likely defaults to the user's locale for rendering purposes. In this case, this looks like we want the data to be based of our HQ/Marketing TZ. This is a copy/paste from the current marketing setup

**Comment by francoislehoux-okta** [comment]
Fixed a bunch

**Review by shawyu-okta — APPROVED** [review]
revert FF changes and 🚀 

🐰

**Pull request merged**